server/controllers/finance/cash.js

```diff
@@ -0,0 +1,247 @@
+/**
+* Cash Controller
+*
+* This controller is responsible for processing cash payments for patients. The
+* payments can either be against an previous invoice (sale payment) or a future
+* invoice (cautionary payment).
+*
+* In order to reduce the burden of accounting on the user, the user will first
+* select a cashbox which implicitly bundles in cash accounts for all supported
+* currencies.  The API accepts a cashbox ID during cash payment creation and
+* looks up the correct account based on the cashbox_id + currency.
+*/
+var db   = require('../../lib/db');
+var uuid = require('../../lib/guid');
+
+// check if an object is empty
+// TODO - can this be from a library?
+function empty(object) {
```

Comment:
The server side API should not do validation and checking, that should be handled at the server side